### PR TITLE
feat(product): add Magento 2 Product Driver

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
This adds the base product driver with the new graphql transformation and query manager architecture. It also restructures the types on the product service to be `DaffProductUnion` or the generic `<T extends DaffProduct>`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/product/src/drivers/magento/transforms/product-transformer.service.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  transform(response: any): DaffProductUnion {
+    const product = response.products.items[0];
+
+    return {
+      id: product.sku,
+      url: product.url_key,
+      name: product.name,
+      images: [
+        {url: product.image.url, id: "0", label: "My label"},
```
Comment (lderrickable) [inline libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: Looks like this is a remnant that shouldn't be here anymore

**Comment by lderrickable** [comment]
Looks like this is real close to complete, so I'll leave it to you to finish it

File: libs/product/src/drivers/magento/transforms/product-transformer.service.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: done.

File: apps/demo/src/app/product/pages/product-view/product-view.component.spec.ts [hunk 2]
```diff
@@ -115,45 +106,54 @@ describe('ProductViewComponent', () => {
...
+      component.ngOnInit();
+      let expected = cold('ab', { a: false, b: true });
+      expect(component.loading$).toBeObservable(expected);
     });
-  });
+
+    it('should dispatch a  DaffProductLoad with an `id`', () => {
+      spyOn(facade, 'dispatch');
+      component.ngOnInit();
+      expect(facade.dispatch).toHaveBeenCalled();
```
Comment (lderrickable) [inline apps/demo/src/app/product/pages/product-view/product-view.component.spec.ts]: should be `expect(facade.dispatch).toHaveBeenCalledWith(new DaffProductLoad(id))`

File: libs/product/src/drivers/magento/transforms/product-transformer.service.ts [hunk 3]
```diff
@@ -10,13 +10,13 @@ export class DaffMagentoProductTransformerService implements DaffProductTransfor
 
   transform(response: any): DaffProductUnion {
     const product = response.products.items[0];
-
+    console.log(product);
```
Comment (lderrickable) [inline libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: remove console.log

**Comment by lderrickable** [comment]
You have a few `xit`s. Not sure if you mean to have those or not. Looks like it might've been on purpose

File: libs/product/src/drivers/magento/transforms/product-transformer.service.ts (hunk 3 above)
Comment (damienwebdev) [reply libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: done.

Comment (lderrickable) [reply libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: I don't see any new commits

Comment (damienwebdev) [reply libs/product/src/drivers/magento/transforms/product-transformer.service.ts]: 73fe4f8e64fc75cad6d517018807d8071d6f2b7b

File: apps/demo/src/app/product/pages/product-view/product-view.component.spec.ts (hunk 2 above)
Comment (damienwebdev) [reply apps/demo/src/app/product/pages/product-view/product-view.component.spec.ts]: 9744558c75652fc3a4b171b5e1bad131e618874d

**Pull request merged**